spec/03-plugins/10-key-auth/01-api_spec.lua

```diff
@@ -310,4 +311,117 @@ describe("Plugin: key-auth (API)", function()
       assert.equal(key_name, body.config.key_names[1])
     end)
   end)
+  describe("/Key-auth", function()
+    local consumer2
+    describe("GET", function()
+      setup(function()
+        helpers.dao:truncate_table("keyauth_credentials")
+        for i = 1, 3 do
+          assert(helpers.dao.keyauth_credentials:insert {
+            consumer_id = consumer.id
+          })
+        end
+        consumer2 = assert(helpers.dao.consumers:insert {
+          username = "bob-the-buidler"
+        })
+        for i = 1, 3 do
+          assert(helpers.dao.keyauth_credentials:insert {
+            consumer_id = consumer2.id
+          })
+        end
+      end)
+      it("retrieve all the key-auths", function()
+        local res = assert(admin_client:send {
+          method = "GET",
+          path = "/key-auth"
+        })
+        local body = assert.res_status(200, res)
+        local json = cjson.decode(body)
+        assert.is_table(json.data)
+        assert.equal(6, #json.data)
+        assert.equal(6, json.total)
+      end)
+      it("retrieve key-auths for a consumer_id", function()
+        local res = assert(admin_client:send {
+          method = "GET",
+          path = "/key-auth?consumer_id=" .. consumer.id
+        })
+        local body = assert.res_status(200, res)
+        local json = cjson.decode(body)
+        assert.is_table(json.data)
+        assert.equal(3, #json.data)
+        assert.equal(3, json.total)
+      end)
+      it("retrieve key-auths for a consumer_id", function()
+        local res = assert(admin_client:send {
+          method = "GET",
+          path = "/key-auth",
+          body = {
+            consumer_id = consumer.id
```

Comment:
I think this might be the same test case as above (`"/key-auth?consumer_id=" .. consumer.id`) with a different syntax allowed by lua-resty-http/our spec helpers? If not, then GET requests shouldn't contain a body anyways. Better remove this test-case regardless: it's either a duplicate of the above one, or a use-case we don't wish to support (and that should be fixed and tested in lua-resty-http or our helpers functions)